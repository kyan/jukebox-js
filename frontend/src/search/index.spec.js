import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import SearchContainer from './index'

describe('SearchContainer', () => {
  const mockStore = configureMockStore()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    it('actions', () => {
      const store = mockStore({
        search: {
          activePage: 2,
          limit: 20,
          query: 'happy',
          total: 3200,
          results: [{
            track: {
              name: 'Track name 1',
              uri: 'track1',
              artist: {
                name: 'Artist name 1'
              },
              album: {
                name: 'Album name 1'
              },
              image: 'image1'
            }
          }],
          searchSideBarOpen: true
        }
      })
      const wrapper = mount(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )
      const actions = store.getActions()
      const event = { target: { value: 'spam' } }
      wrapper.find('SidebarPusher').simulate('click')
      wrapper.find('Form').simulate('submit')
      wrapper.find('.search-list-item').first().simulate('click')
      wrapper.find('input').first().simulate('change', event)
      wrapper.find('Pagination').find('.item').last().simulate('click')
      expect(actions).toMatchSnapshot()
    })
  })
})
