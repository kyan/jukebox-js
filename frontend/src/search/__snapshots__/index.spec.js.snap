// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer render actions 1`] = `
Array [
  Object {
    "open": false,
    "type": "actionToggleSearchSidebar",
  },
  Object {
    "key": "search::getTracks",
    "params": Object {
      "options": Object {
        "limit": 20,
        "offset": 0,
      },
      "query": "happy",
    },
    "type": "actionTrackSearch",
  },
  Object {
    "type": "actionRemoveFromSearchResults",
    "uri": "track1",
  },
  Object {
    "key": "mopidy::tracklist.add",
    "params": Object {
      "uri": null,
    },
    "type": "actionSend",
  },
  Object {
    "query": "spam",
    "type": "actionStoreSearchQuery",
  },
  Object {
    "key": "search::getTracks",
    "params": Object {
      "options": Object {
        "limit": 20,
        "offset": 20,
      },
      "query": "happy",
    },
    "type": "actionTrackSearch",
  },
]
`;
